src/test/java/edu/ucsb/cs156/organic/controllers/CoursesControllerTests.java

```diff
@@ -0,0 +1,327 @@
+import static org.awaitility.Awaitility.await;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.AutoConfigureDataJpa;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Import;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.web.servlet.MvcResult;
+
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+import static org.springframework.security.test.web.servlet.request.SecurityMockMvcRequestPostProcessors.csrf;
+
+import edu.ucsb.cs156.organic.entities.Course;
+import edu.ucsb.cs156.organic.entities.Staff;
+import edu.ucsb.cs156.organic.entities.User;
+import edu.ucsb.cs156.organic.entities.jobs.Job;
+import edu.ucsb.cs156.organic.repositories.CourseRepository;
+import edu.ucsb.cs156.organic.repositories.StaffRepository;
+import edu.ucsb.cs156.organic.repositories.UserRepository;
+import edu.ucsb.cs156.organic.repositories.jobs.JobsRepository;
+import edu.ucsb.cs156.organic.services.jobs.JobService;
+import lombok.extern.slf4j.Slf4j;
+
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+
+@Slf4j
+@WebMvcTest(controllers = CoursesController.class)
+@Import(JobService.class)
+@AutoConfigureDataJpa
+public class CoursesControllerTests extends ControllerTestCase {
+
+        @MockBean
+        UserRepository userRepository;
+
+        @MockBean
+        CourseRepository courseRepository;
+
+        @MockBean
+        StaffRepository courseStaffRepository;
+
+        @Autowired
+        ObjectMapper objectMapper;
+
+        Course course1 = Course.builder()
+                        .id(1L)
+                        .name("CS156")
+                        .school("UCSB")
+                        .term("F23")
+                        .start(LocalDateTime.parse("2023-09-01T00:00:00"))
+                        .end(LocalDateTime.parse("2023-12-31T00:00:00"))
+                        .githubOrg("ucsb-cs156-f23")
+                        .build();
+
+        Course course2 = Course.builder()
+                        .id(1L)
+                        .name("CS148")
+                        .school("UCSB")
```

Comment:
For the example given for school name you use lowercase ucsb and here you use uppercase. more of a nitpick than anything but worth considering

Reply:
noted... I'll do a global check

Reply:
Fixed